src/PostgresEventStore.php

```diff
@@ -81,6 +81,9 @@ public function __construct(
 
         Assertion::min($loadBatchSize, 1);
 
+        $this->setAttribute(PDO::ATTR_STATEMENT_CLASS, ['Doctrine\DBAL\Driver\PDOStatement', []]);
+        $this->setAttribute(PDO::ATTR_ERRMODE, PDO::ERRMODE_EXCEPTION);
```

Comment:
I don't want to force the user to a specific usage here.

Reply:
Can you explain? The PDO is not intended to be used outside of the EventStore, right? So it should be ok to use `PDO::ERRMODE_EXCEPTION`. And honestly noone should ever use PDO without this - the need to check errors manually leads to critical mistakes like #118 which is what I want to prevent.

Reply:
The problem is, that someone with access to the PDO instance, can change this attribute afterwards and the event-store stops working suddenly. The bug has been fixed and I don't expect to get another one popping up because of this. I think we leave it as is.

Reply:
In that case we should create the PDO in constructor instead of require it. I actually do expect furhter problems caused by this. Hopefully not as severe but the exceptions thrown by the EventStore are quite inconsistent now and there is a lot of duplicated code. It does need refactoring.

Reply:
code duplication is not an issue. what do you mean with inconsistency in exceptions? can you show some examples?

Reply:
changing the constructor is a bc break and i will not do this now.